tests/test_platform.py

```diff
@@ -0,0 +1,15 @@
+from contextlib import contextmanager
+from unittest.mock import MagicMock
+
+import builtins
+import xonsh.platform as xp
+
+
+def test_githash_value_error(monkeypatch):
+    @contextmanager
+    def mocked_open(*args):
+        yield MagicMock(read=lambda: 'abc123')
+    monkeypatch.setattr(builtins, 'open', mocked_open)
+    sha, date_ = xp.githash()
+    assert date_ is None
+    assert sha is None
```

Comment:
Tried to add a normal case, but haven't found a way to mock up `functools.lru_cache`. So only bad case added.